### PR TITLE
xDAI and Matic L2 Bridges and messenger wrappers

### PR DESCRIPTION
**Pull request by cwhinfrey**
* Add xDAI L2 Bridge and messenger wrapper

### CODE REVIEW

**Comment by cwhinfrey** [comment]
WIP - Do not merge

Still need to add Matic

**Comment by cwhinfrey** [comment]
There are some complications with Matic that I need to clarify with their team. I'll do Matic as a separate PR so this can be merged if it looks good.

File: contracts/bridges/L2_XDaiBridge.sol [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            supportedChainIds,
+            bonders,
+            exchangeAddress,
+            name,
+            symbol,
+            decimals
+        )
+    {
+        messenger = _messenger;
+        l1ChainId = bytes32(supportedChainIds[0]);
```
Comment (shanefontaine) [inline contracts/bridges/L2_XDaiBridge.sol:41]: @cwhinfrey If we do not pass in mainnet as the first parameter to `supportedChainIds` then I believe this breaks. If that is correct, I think we should add a `require(supportedChainIds[0] === 1);` here.

**Review by shanefontaine — CHANGES_REQUESTED** [review]

File: contracts/bridges/L2_XDaiBridge.sol (hunk 1 above)
Comment (cwhinfrey) [reply contracts/bridges/L2_XDaiBridge.sol]: This is now passed in explicitly

**Review by shanefontaine — APPROVED** [review]
👍

**Pull request merged**